src/test/java/BasicMAPF/Solvers/PIBT/PIBT_SolverTest.java

```diff
@@ -479,20 +479,43 @@ void emptyMapValidityStayInPlaceConstraint() {
         assertEquals(22 , solved.sumServiceTimes());
     }
 
-    // the following test important to check specific scenario where agent reached his goal,
-    // but can't stay in place since there is a infinite constraint, a constraint about his final location in future timestamp
-    // this scenario throws an UnsupportedOperationException
+
     @Test
-    void emptyMapValidityInfiniteConstraintThrowsError() {
+    void emptyMapValidityInfiniteConstraintTest() {
         MAPF_Instance testInstance = instanceEmpty1;
 
         I_Coordinate coor02 = new Coordinate_2D(1,2);
         Constraint constraint1 = new Constraint(agent33to12, 10, mapEmpty.getMapLocation(coor02));
         ConstraintSet constraints = new ConstraintSet();
         constraints.add(constraint1);
 
-        assertThrows(UnsupportedOperationException.class, () -> {
-            PIBT_Solver.solve(testInstance, new RunParametersBuilder().setTimeout(timeout).setConstraints(constraints).setInstanceReport(instanceReport).createRP());
-        });
+        Solution solved = PIBT_Solver.solve(testInstance, new RunParametersBuilder().setTimeout(timeout).setConstraints(constraints).setInstanceReport(instanceReport).createRP());
+        System.out.println(solved.readableToString());
+        assertTrue(solved.solves(testInstance));
+        assertEquals(36, solved.sumIndividualCosts());
+        assertEquals(8, solved.makespan());
+        assertEquals(22 , solved.sumServiceTimes());
+    }
+
+    @Test
+    void emptyMapValidityInfiniteConstraintTestBothAgents() {
+        MAPF_Instance testInstance = instanceCircle1;
+
+        I_Coordinate coor33 = new Coordinate_2D(3,3);
+        I_Coordinate coor12 = new Coordinate_2D(1,2);
```

Comment:
Not so important, but you can import these from `TestConstants.Coordinates` instead of generating them from scratch